API: add antialiased to interpolation-stage in image (#28061)
matplotlib/matplotlib: doc/api/next_api_changes/behavior/28061-JMK.rst

imshow *interpolation_stage* default changed to 'antialiased'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The *interpolation_stage* keyword argument `~.Axes.imshow` has a new default

> value 'antialiased'.  For images that are down-sampled or up-sampled less than
> a factor of three, image interpolation will occur in 'rgba' space.  For images

Reviewer: ```suggestion
value 'antialiased'.  For images that are up-sampled less than a factor of 
three or down-sampled , image interpolation will occur in 'rgba' space.  For images
```

I read it as "(down- or up-sampled) by less than a factor of 3" which is not what is meant.

that are up-sampled by more than a factor of 3, then image interpolation occurs
in 'data' space.

The previous default was 'data', so down-sampled images may change subtly with
the new default.  However, the new default also avoids floating point artifacts
at sharp boundaries in a colormap when down-sampling.

> The previous behavior can achieved by changing :rc:`image.interpolation_stage`.

Reviewer 2: ```suggestion
The previous behavior can achieved by setting :rc:`image.interpolation_stage` or
the *interpolation_stage* parameter in `~.axes.Axes.imshow` to 'data'.
```

imshow *interpolation_stage* default changed to 'auto'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The *interpolation_stage* keyword argument `~.Axes.imshow` has a new default
value 'auto'.  For images that are up-sampled less than a factor of
three or down-sampled, image interpolation will occur in 'rgba' space.  For images

> that are up-sampled by more than a factor of 3, then image interpolation occurs

Reviewer 2: ```suggestion
that are up-sampled by a factor of 3 or more, then image interpolation occurs
```
